### ISSUE TITLE
electron-webpack does in fact impose or encourage a specific project structure

### ISSUE THREAD

**Issue by ThaJay** [electron-userland/electron-webpack#152]
>  not impose or encourage any sort of project structure
https://webpack.electron.build/core-concepts

- the renderer js file needs to sit in a folder ('renderer' per default), project root is not allowed.
- the main electron js file needs to sit in a folder ('main' per default), project root is not allowed.
- webpack-dev-server only watches _project-root_/src and not anything else, instead of the normal webpack behaviour to follow imports / watch the whole project folder.

I used to have a `index.js` and `renderer.js` in my project root. Also in my project root were 2 submodules used by other react projects. All this has been moved to the speciefied locations in `/src` and now it all works with minimal config. The configuration options provided in the docs seem to be not advanced enough to work as expected.

**Comment by deadcoder0904** [comment]
This is true. It is a false claim to say it does not impose or encourage any sort of project structure :)

**Comment by JM-Mendez** [comment]
The documentation clearly states that the only requirement is for the main and renderer entry points to be in separate directories. After that there is no structure enforced as far as I'm aware.

[project-structure.md#webpack-entry-points](https://github.com/electron-userland/electron-webpack/blob/master/docs/en/project-structure.md#webpack-entry-points)
> One important thing to notice is that electron-webpack expects that your main process and renderer process source code is separated using different directories. Within each process's directory, one of the following entry files is expected...
>
>index.js / main.js
index.ts / main.ts (when using the TypeScript add-on)
The entry files are the main starting point of your webpack bundle, so everything found within its dependency tree will be bundled.

**Comment by deadcoder0904** [comment]
> The documentation clearly states that the only requirement is for the main and renderer entry points to be in separate directories.

This clearly means that it imposes. Idk how to explain this better but even if you mention one condition then it means that it imposes. There is no only in this :)

**Comment by JM-Mendez** [comment]
That's just being pedantic IMHO. By your definition no software or computing language will ever qualify as 'not imposing', since even an api requires some structure. I mean, we don't claim our boss imposes job requirements on us, right? There's a difference between a requirement and an imposition.

As far as software is concerned, there isn't any real imposition here. Just 2 relatively minor requirements - that your `main.(js|ts)` files live in two separate folders. That's it. After that you're free to place files wherever. 

So instead of putting the actual main files in the project root, you could put a main and renderer folder with just those two files. Then throw all your other files in the project root if you wanted.

Hell, you may even be able to symlink them from your project root if you really demanded that structure.

Compare that to something like nextjs, where you **must** put all your main pages in a folder named explicitly `pages`. Or with their new api having files with specific names like `_app.js` or `_document.js`. Have fun symlinking entire directories to meet those requirements.

But tomato, tomatoh I suppose. ¯\_(ツ)_/¯

**Comment by deadcoder0904** [comment]
Well I guess we're gonna have to agree to disagree then. And No, not every software imposes some requirement. For example, for a simple Node app you can put your files wherever you want. So bottomline, I think its imposing. You are welcome to have your own opinion but for me it definitely is imposing :)

**Comment by ThaJay** [comment]
I have nothing to add to the discussion. Fact is I had to significantly change the folder structure of my project. It makes navigation using the commandline slightly less convenient (like having a submodule views and a folder viewcomponents both in the src folder). This is still acceptable to me but that sentence in the documentation is plain wrong.

**Comment by ThaJay** [comment]
Read the opening post and you can see; no.
